### PR TITLE
Feat(eos_designs): macsec support fabric links #1506

### LINKED ISSUE #1506: Feat(eos_designs): Include MACSec support for fabric links
### Enhancement summary

MACSec should be configurable for inter-fabric links

### Which component of AVD is impacted

eos_designs

### Use case example

Allow MACSec between leaves and spines.

### Describe the solution you would like

Variable or signalisation that will allow eos_designs to enable MACSec in a set of defined links, not all at the same time. 

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ankudinov**
## Change Summary

Introduce a node key to enable MacSec on all uplinks.

## Related Issue(s)

Fixes #1506 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Following key will be added uplink management and follow the same logic as uplink_ptp, etc.

```yaml
< node_type_key >:

  defaults:
    # Enable MacSec on all uplinks | Optional
    uplink_macsec:
      profile: "< MacSec profile name >"
```

## How to test
molecule converge --scenario-name eos_designs-twodc-5stage-clos

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md [hunk 1]
```diff
@@ -56,13 +56,13 @@ pod_name: < POD_Name >
 
 The following table provide information on the default node types that have been pre-defined in [`eos_designs/defaults/main/defaults-node-type-keys.yml`](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/roles/eos_designs/defaults). To customize or create new node types, please refer to [node types definition](node-types.md)
 
-| Node Type Key      | Underlay Router | Uplink Type | Default EVPN Role  | L2 Network Services | L3 Network Services | VTEP | MLAG Support | Connected Endpoints |
+| Node Type Key      | Underlay Router | Uplink Type  | Default EVPN Role | L2 Network Services | L3 Network Services | VTEP | MLAG Support | Connected Endpoints |
```
Comment (xaviramon) [inline ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md]: Is it really needed to change all this?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md]: Please avoid "fixing" unrelated stuff in a feature PR.

File: ansible_collections/arista/avd/roles/eos_designs/templates/facts/topology/p2p-uplinks.j2 [hunk 2]
```diff
@@ -25,6 +25,10 @@ topology:
 {%             if switch.uplink_ptp.enable is arista.avd.defined %}
       ptp: {{ switch.uplink_ptp.enable }}
 {%             endif %}
+{%             if switch.uplink_macsec.profile is arista.avd.defined %}
+      mac_security:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/facts/topology/p2p-uplinks.j2]: you could just do `mac_security: {{ switch.uplink_macsec }}` so any other future options are caried along.
Adjust the `if` above to just test for `uplink_macsec is arista.avd.defined`

File: ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md [hunk 3]
```diff
@@ -203,6 +203,10 @@ defaults <- node_group <- node_group.node <- node
     uplink_ptp:
       enable: < boolean >
 
+    # Enable MacSec on all uplinks | Optional
+    uplink_macsec:
+      profile: "< MacSec profile name >"
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md:208]: Would we need a datamodel for defining the macsec profile itself? Or can we just use the `eos_cli_config_gen` one directly?

File: ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/inventory/group_vars/DC1_POD1.yml [hunk 4]
```diff
@@ -8,30 +8,33 @@ spine:
     platform: vEOS-LAB
     bgp_as: "65110.100"
     loopback_ipv4_pool: 172.16.110.0/24
-    uplink_interfaces: ['Ethernet1', 'Ethernet2']
-    uplink_switches: ['DC1-SUPER-SPINE1', 'DC1-SUPER-SPINE2']
+    uplink_interfaces: ["Ethernet1", "Ethernet2"]
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/inventory/group_vars/DC1_POD1.yml:11]: All these other changes only makes it harder to review. It doesn't really matter if we use single or double quotes or mix them. This is just for molecule. Same regarding spacing inside lists.

File: ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md (hunk 3 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md:208]: I believe it's fine to define the profile directly with eos_cli_config_gen. This is similar to current `p2p_uplinks_qos_profile: < qos_profile_name >`

File: ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md (hunk 1 above)
Comment (ankudinov) [reply ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md]: @ClausHolbechArista @xaviramon this is done by linter that is part of AVD devcontainer. It wasn't a manual change.

File: ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/inventory/group_vars/DC1_POD1.yml (hunk 4 above)
Comment (ankudinov) [reply ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/inventory/group_vars/DC1_POD1.yml:11]: same as above, that is not a manual change

File: ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md:208]: ok

File: ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md]: I am not sure which container or linter that is. We don't have this linting in our CI.
Please avoid adding these changes that are unrelated to the feature.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/fabric-topology.md]: As discussed, we will accept these for now and decide what we will do in the future on the next dev call.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by xaviramon — APPROVED** [review]
LGTM

**Pull request merged**